app/src/main/python/gogdl/dl/dl_utils.py

```
@@ -0,0 +1,184 @@
+def get_secure_link(api_handler, path, gameId, generation=2, logger=None, root=None):
+    url = ""
+    if generation == 2:
+        url = f"{constants.GOG_CONTENT_SYSTEM}/products/{gameId}/secure_link?_version=2&generation=2&path={path}"
+    elif generation == 1:
+        url = f"{constants.GOG_CONTENT_SYSTEM}/products/{gameId}/secure_link?_version=2&type=depot&path={path}"
+    if root:
+        url += f"&root={root}"
+
+    try:
+        r = requests.get(url, headers=api_handler.session.headers, timeout=TIMEOUT)
+    except BaseException as exception:
```

Ruff (0.12.2)

Do not catch blind exception: `BaseException`

(BLE001)